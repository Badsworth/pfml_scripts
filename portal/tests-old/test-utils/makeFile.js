/**
 * Make a File instance
 * @see https://developer.mozilla.org/en-US/docs/Web/API/File
 * @param {object} attrs File attributes
 * @param {string} attrs.name The file's name
 * @param {string} attrs.type The file's MIME type
 * @returns {File}
 */
export const makeFile = (attrs = {}) => {
  const { name, type } = Object.assign(
    {
      name: "file.pdf",
      type: "application/pdf",
    },
    attrs
  );

  return new File([], name, { type });
};

/**
 * Create an HTML <input> for a test's event.target
 * @param {object} attrs - HTML attributes for the input
 * @returns {HTMLInputElement}
 */
export function createInputElement(attrs) {
  const input = document.createElement("input");

  Object.entries(attrs).forEach(([attributeName, value]) => {
    input.setAttribute(attributeName, value);
  });

  if (attrs.checked === false) input.checked = false;

  return input;
}
