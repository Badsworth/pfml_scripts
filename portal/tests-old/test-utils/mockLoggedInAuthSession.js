import { Auth } from "@aws-amplify/auth";

/**
 * Mock the Cognito authentication session so a test user appears logged in
 */
const mockLoggedInAuthSession = () => {
  const accessTokenJwt =
    "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1lIjoiQnVkIn0.YDRecdsqG_plEwM0H8rK7t2z0R3XRNESJB5ZXk-FRN8";

  jest.spyOn(Auth, "currentSession").mockImplementation(() =>
    Promise.resolve({
      accessToken: { jwtToken: accessTokenJwt },
    })
  );
};

export default mockLoggedInAuthSession;
