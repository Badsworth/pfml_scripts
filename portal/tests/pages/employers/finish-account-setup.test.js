import { screen, waitFor } from "@testing-library/react";
import FinishAccountSetup from "../../../src/pages/employers/finish-account-setup";
import { renderPage } from "../../test-utils";
import userEvent from "@testing-library/user-event";

describe("FinishAccountSetup", () => {
  it("renders the page", () => {
    const { container } = renderPage(FinishAccountSetup, { isLoggedIn: false });
    expect(container).toMatchSnapshot();
  });

  it("calls forgotPassword when the form is submitted", async () => {
    const spy = jest.fn();
    renderPage(FinishAccountSetup, {
      addCustomSetup: (appLogic) => {
        appLogic.auth.forgotPassword = spy;
      },
      isLoggedIn: false,
    });

    const email = "email@test.com";
    userEvent.type(screen.getByLabelText(/Email/i), email);
    userEvent.click(screen.getByRole("button", { name: /Submit/i }));

    await waitFor(() => {
      expect(spy).toHaveBeenCalledWith(email);
    });
  });
});
