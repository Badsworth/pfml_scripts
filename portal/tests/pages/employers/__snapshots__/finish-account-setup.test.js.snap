// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinishAccountSetup renders the page 1`] = `
<form
  className="usa-form"
  onSubmit={[Function]}
>
  <Title>
    Finish setting up your account
  </Title>
  <Lead>
    Please verify your email address. If an account exists for the email you provide, you’ll receive a 6-digit verification code. After that, you’ll be prompted to create a new password.
  </Lead>
  <InputText
    label="Email address"
    name="username"
    onChange={[Function]}
    smallLabel={true}
    type="email"
    value=""
  />
  <Button
    loading={false}
    type="submit"
  >
    Submit
  </Button>
  <div
    className="margin-top-2 text-bold"
  >
    <Link
      href="/employers/create-account"
    >
      <a>
        Create a new account
      </a>
    </Link>
  </div>
</form>
`;
